### ISSUE TITLE
Parameterize RC pipeline

### ISSUE THREAD

**Issue by jdpsen** [primeqa/primeqa#192]
_aferritto created the following on Feb 22:_  

 Add parameters for lm, lr, etc for rc pipeline 

_Original issue: https://github.ibm.com/ai-foundation/PrimeQA/issues/41_

**Comment by jdpsen** [comment]
_On Mar 29, zhangr commented:_
HF transformers scripts generally use 3 classes to process parameters:
#### ModelArguments, DataTrainingArguments and TrainingArguments. ####
(https://github.com/huggingface/transformers/blob/main/examples/pytorch/question-answering/run_qa.py)

```
parser = HfArgumentParser((ModelArguments, DataTrainingArguments, TrainingArguments)) 
if len(sys.argv) == 2 and sys.argv[1].endswith(".json"):
        # If we pass only one argument to the script and it's the path to a json file,
        # let's parse it to get our arguments.
        model_args, data_args, training_args = parser.parse_json_file(json_file=os.path.abspath(sys.argv[1]))
else:
        model_args, data_args, training_args = parser.parse_args_into_dataclasses()
```

**Comment by jdpsen** [comment]
_On Mar 29, zhangr commented:_
#### TrainingArguments #### 
(https://github.com/huggingface/transformers/blob/main/src/transformers/training_args.py)

This is the class that handles most of the parameters used in training/eval/test. It's also part of the transformers, so it can be imported to our script.

parameters supported:
output_dir, do_train, do_eval, do_predict, per_device_train_batch_size, per_device_eval_batch_size, gradient_accumulation_steps, learning_rate, weight_decay, adam_beta1, adam_beta2, adam_epsilon, max_grad_norm, num_train_epochs, lr_scheduler_type, warmup_steps, log_level, seed, fp16, local_rank, dataloader_num_workers, resume_from_checkpoint, metric_for_best_model, save_strategy, save_steps,
......

**Comment by jdpsen** [comment]
_On Mar 29, zhangr commented:_
#### ModelArguments ####
(https://github.com/huggingface/transformers/blob/main/examples/pytorch/question-answering/run_qa.py)

model_name_or_path
config_name
tokenizer_name
cache_dir
......


#### DataTrainingArguments ####
(https://github.com/huggingface/transformers/blob/main/examples/pytorch/question-answering/run_qa.py)

dataset_name
dataset_config_name
train_file
validation_file
test_file
preprocessing_num_workers
max_seq_length
pad_to_max_length
null_score_diff_threshold
doc_stride
n_best_size
max_answer_length
max_train_samples
max_eval_samples
......

**Comment by jdpsen** [comment]
_On Mar 29, zhangr commented:_
#### OneQA specific ####

scorer_type
task_heads
preprocessor_class
data_collator (DataCollatorWithPadding ???)
postprocessor_class
metrics_fn
sampling_rate (???)
initial_model_weights (???)

**Comment by jdpsen** [comment]
_On Apr 05, jaydesen commented:_
PR reviewed and need some revision.

**Comment by jdpsen** [comment]
This is done, code merged into PrimeQA main.
